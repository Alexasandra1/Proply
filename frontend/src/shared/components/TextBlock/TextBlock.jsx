import React from 'react';
import './TextBlock.scss';

export function TextBlock(props){
  return (
    <div className={`text-block ${props.styleTextBlock}`}>
      <h1 className={`title ${props.styleTitle}`} >{props.title}</h1>
      <p className={`text ${props.styleText}`}>{props.text}</p>
    </div>
  );
}

