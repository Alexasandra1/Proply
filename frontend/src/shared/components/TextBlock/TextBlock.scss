.text-block {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  color: #000;
  font-family: Montserrat;
}

.text-block_statistick {
  width: 260px;
  height: 135px;
}

.title_main {
  font-size: 40px;
  font-weight: 700;
  margin-bottom: 15px;
  margin-top: 0;
}

.title_statistick {
  font-size: 55px;
  text-align: left;
  font-weight: bolder;
  margin-top: 0;
  margin-bottom: 10px;
}

.text_statistick {
  font-size: 20px;
  text-align: left;
}

.text_main {
  font-size: 20px;
  font-weight: 400;
  text-align: left;
  margin: 0 0 40px 0;
}

.title_about {
  text-align: left;
  font-size: 96px;
  font-weight: 700;
  //margin-bottom: 30px;
}


.text_about {
  font-size: 30px;
  font-weight: 400;
  text-align: left;
}

.textBlockMain {
  max-width: 51rem;
  text-align: center;
  align-items: center;
}

.styleTitleJoin {
  font-family: Montserrat;
  font-size: 6rem;
  font-style: normal;
  font-weight: 700;
}

.styleTextJoin {
  text-align: center;
  font-family: Montserrat;
  font-size: 2rem;
  font-style: normal;
  font-weight: 400;
}

.title_cons {
  color: #000;
  font-family: Montserrat;
  font-size: 3.4375rem;
  font-style: normal;
  font-weight: 700;
  margin: 0 0 1rem 0;
}

.text_consFisrt {
  color: #000;
  font-family: Montserrat;
  font-size: 2.25rem;
  font-style: normal;
  font-weight: 500;
  margin: 0;
}

.text__consSecond{
  color: #000;
  font-family: Montserrat;
  margin: 0;
font-size: 1.25rem;
font-style: normal;
font-weight: 500;
text-align: left;
}

.text__consThird{
  color: #000;
  font-family: Montserrat;
  font-size: 1.5rem;
  font-style: normal;
  font-weight: bolder;
  margin: 0;
}



@media (max-width: 740px) {
  .text-block {
    justify-content: center;
  }
  .title_cons {
    font-size: 29px;
  }
}

@media (max-width: 330px) {
  .title_main {
    font-size: 24px;
    font-weight: 700;
  }

  .text_main {
    font-size: 16px;
    font-weight: 400;
  }

  .title_about {
    font-size: 48px;
    font-weight: 700;
    //margin-bottom: 30px;
  }


  .text_about {
    font-size: 20px;
    font-weight: 400;
  }

}