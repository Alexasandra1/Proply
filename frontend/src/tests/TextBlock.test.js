import React from 'react';
import renderer from 'react-test-renderer';
import { TextBlock } from '../shared/components/TextBlock/TextBlock';

describe('TextBlock', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<TextBlock styleTextBlock="style" title="title" styleTitle="style" text="text" styleText="style" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
