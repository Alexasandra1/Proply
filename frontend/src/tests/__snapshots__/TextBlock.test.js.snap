// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBlock renders correctly 1`] = `
<div
  className="text-block style"
>
  <h1
    className="title style"
  >
    title
  </h1>
  <p
    className="text style"
  >
    text
  </p>
</div>
`;
